test/fiddle/test_pointer.rb

```diff
@@ -10,6 +10,38 @@ def dlwrap arg
       Fiddle.dlwrap arg
     end
 
+    def test_can_write_memory
+      # Allocate some memory
+      address = Fiddle.malloc(Fiddle::SIZEOF_VOIDP)
+
+      bytes_to_write = Fiddle::SIZEOF_VOIDP.times.to_a.pack("C*")
+
+      # Write to the memory
+      Fiddle::Pointer.write(address, bytes_to_write)
+
+      # Read the bytes out again
+      bytes = Fiddle::Pointer.read(address, Fiddle::SIZEOF_VOIDP)
+      assert_equal bytes_to_write, bytes
+    ensure
+      Fiddle.free address
+    end
+
+    def test_can_read_memory
+      # Allocate some memory
+      address = Fiddle.malloc(Fiddle::SIZEOF_VOIDP)
+
+      # Write some bytes in to the memory
+      ptr = Fiddle::Pointer.new(address)
+      bytes_to_write = Fiddle::SIZEOF_VOIDP.times.to_a
+      bytes_to_write.each { |i| ptr[i] = i }
+
+      # Read the bytes out again
+      bytes = Fiddle::Pointer.read(address, Fiddle::SIZEOF_VOIDP).bytes
+      assert_equal bytes_to_write, bytes
+    ensure
+      Fiddle.free address
+    end
```

Comment:
How about unifying them? It seems that the first test covers the second one.

```suggestion
    def test_can_read_write_memory
      # Allocate some memory
      address = Fiddle.malloc(Fiddle::SIZEOF_VOIDP)

      bytes_to_write = Fiddle::SIZEOF_VOIDP.times.to_a.pack("C*")

      # Write to the memory
      Fiddle::Pointer.write(address, bytes_to_write)

      # Read the bytes out again
      bytes = Fiddle::Pointer.read(address, Fiddle::SIZEOF_VOIDP)
      assert_equal bytes_to_write, bytes
    ensure
      Fiddle.free address
    end
```